tests/meson.build

```diff
@@ -0,0 +1,74 @@
+# Test sources
+test_src = files(
+    'test_CommChannel.cc',
+    'test_Context.cc',
+    'test_exceptions.cc',
+    'test_Executor.cc',
+    #'tests/test_format.cc' needs fmt{} library
+    'test_LockedQueue.cc',
+    'test_main.cc',
+    'test_Message.cc',
+    'test_Sequence.cc',
+    'test_SequenceManager.cc',
+    'test_serialize_sequence.cc',
+    'test_Step.cc',
+    'test_time_types.cc',
+    'test_VariableName.cc',
+)
+
+# The tests are executed in the build dir to avoid pollution of the git repository with
+# the test executable's output files
+test('all',
+    executable(meson.project_name() + '_test',
+        test_src,
+        dependencies : taskolib_dep,
+    ),
+    workdir : meson.current_build_dir(),
+    timeout : 10,
+)
+
+fmt_dep = dependency('fmt', required : false)
+if fmt_dep.found()
+    test('format',
+        executable(meson.project_name() + '_format_test',
+            [ 'test_format.cc', 'test_main.cc', ],
+            dependencies : [ taskolib_dep, dependency('fmt'), ],
+        ),
+        timeout : 10,
+    )
+endif
+
+######
+# Test if all headers are self-contained (Core Guidelines SF.11)
```

Comment:
You mean [SF.11: Header files should be self-contained](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf11-header-files-should-be-self-contained) on ISO Cpp?

Reply:
Yes, the C++ Core Guidelines.